src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1735,6 +1736,35 @@ source = do
 linkTitle :: PandocMonad m => MarkdownParser m Text
 linkTitle = quotedTitle '"' <|> quotedTitle '\''
 
+-- Github wiki style link, with optional title
+-- syntax documented under https://help.github.com/en/github/building-a-strong-community/editing-wiki-content
+githubWikiLink :: PandocMonad m => MarkdownParser m (F Inlines)
+githubWikiLink = try $ do
+  guardEnabled Ext_wikilink
+  string "[["
+  contents <- lookAhead $ manyTillChar anyChar (try $ string "]]")
+  let title = "wikilink"
+  if T.any (== '|') contents
+  then do
+    st <- getState
+    setState $ st{ stateAllowLinks = False }
+    label <- mconcat <$> manyTill inline (char '|')
+    setState $ st{ stateAllowLinks = True }
+    url <- manyTillChar anyChar $ try $ string "]]"
+    return $ B.link url title <$> label
```

Comment:
This approach seems potentially unreliable.  Example: 
```
[[hello `|` there]]
```
IN this case `contents` will be `` hello `|` there ``, but when we do `manyTill inline (char '|')` we will keep parsing past the `]]`, because the `|` character occurs *within* an inline (a code span).

Reply:
This is tricky, and I'm not sure how it should behave, because we don't know whether to treat the content as plain text or inlines until we know whether there's a `|` character, but that `|` character might be part of regular inline content.  I'm really not sure what to do here.

Reply:
Good point, I have actually borrowed it from the VimWiki parser, we may have the same issue.